cpmpy/tools/xcsp3/xcsp3_benchmark.py

```diff
@@ -0,0 +1,289 @@
+
+    try:
+        init_signal_handlers() # configure OS signal handlers
+        xcsp3_cpmpy(**kwargs)
+        conn.send({"status": "ok"})
+    except Exception as e: # capture exceptions and report in state
+        tb_str = traceback.format_exc()
+        conn.send({"status": "error", "exception": e, "traceback": tb_str})
+    finally:
+        sys.stdout = original_stdout
+        conn.close()
+
+# exec_args = (filename, metadata, solver, time_limit, mem_limit, output_file, verbose) 
+def execute_instance(args: Tuple[str, dict, str, int, int, str, bool]) -> None:
+    """
+    Solve a single XCSP3 instance and write results to file immediately.
+    
+    Args is a list of:
+        filename: Path to the XCSP3 instance file
+        metadata: Dictionary containing instance metadata (year, track, name)
+        solver: Name of the solver to use
+        time_limit: Time limit in seconds
+        mem_limit: Memory limit in MB
+        output_file: Path to the output CSV file
+        verbose: Whether to show solver output
+    """
+    warnings.filterwarnings("ignore")
+    
+    filename, metadata, solver, time_limit, mem_limit, output_file, verbose = args
+
+    # Fieldnames for the CSV file
+    fieldnames = ['year', 'track', 'instance', 'solver',
+                  'time_total', 'time_parse', 'time_model', 'time_post', 'time_solve',
+                  'status', 'objective_value', 'solution']
+    result = dict.fromkeys(fieldnames)  # init all fields to None
+    result['year'] = metadata['year']
+    result['track'] = metadata['track']
+    result['instance'] = metadata['name'] 
+    result['solver'] = solver
+
+    # Decompress before timers start
+    if str(filename).endswith(".lzma"):
+        # Decompress the XZ file
+        with lzma.open(filename, 'rt', encoding='utf-8') as f:
+            xml_file = StringIO(f.read()) # read to memory-mapped file
+            filename = xml_file
+            
+    # Start total timing
+    total_start = time.time()
+
+    # Call xcsp3 in separate process
+    ctx = multiprocessing.get_context("spawn")
+    parent_conn, child_conn = multiprocessing.Pipe() # communication pipe between processes
+    process = ctx.Process(target=xcsp3_wrapper, args=(child_conn, {"benchname":filename, "solver": solver, "time_limit": time_limit, "mem_limit": mem_limit}, verbose))
+    process.start()
+    process.join(timeout=time_limit)
+
+    # Replicate competition convention on how jobs get terminated
+    if process.is_alive():
+        # Send sigterm to let process know it reached its time limit
+        os.kill(process.pid, signal.SIGTERM)
+        # 1 second grace period
+        process.join(timeout=1)
+        # Kill if still alive
+        if process.is_alive():
+            os.kill(process.pid, signal.SIGKILL)
+            process.join()
+
+    result['time_total'] = time.time() - total_start
+
+    # Collect output
+    output = []
```

Comment:
should we do it this way and collect all output in an explicit array?
Can't we directly do 
````while parent_conn.poll(...):
   line = parent_conn.recv()
   if line.startswith('s ')...
```
and then an extra check on the 'exit state missing' that I don't fully get